rfd/0057-automatic-aws-server-discovery.md

```diff
@@ -235,6 +235,153 @@ discovery_service:
     teleport.dev/origin: "cloud"
 ```
 
+
+### Agentless installation
+
+In addition to supporting automatic Teleport agent installation, an agentless option
+will also be supported. This mode will update the OpenSSH CA to use the Teleport CA
+without installing the full Teleport Agent.
+
+The teleport ca and host keys will be generated and then uploaded to
+the the AWS Secret store where, for each node discovered a new secret
+will be created, each secret will have the following contents
+
+```json
+{
```

Comment:
What about CA rotations? 

In order to not break access to these agentless nodes, the discovery service would need to watch for CA rotations and push updated certs to the agentless nodes.

Are we including this in scope, or are we okay with the fact that CA rotations will break things for now?

Reply:
I don't see why automatic upgrades apply. That's about updating the Teleport binary, not getting new certs.

I'm no expert here, but I would think discovery service could set up a watcher for CA rotations just like Machine ID does, and push the new certs to the instances when this happens.

Reply:
I didn't mean automatic upgrades specifically but rather machinery that's part of script-based upgrades for example, basically the machinery that will run some script on the nodes based on some conditions (new version availability is just one of the conditions in my mind). But yeah, it could be done in discovery service too.